### DISCUSSION TITLE
Any way to get motion events from the camera?

### DISCUSSION THREAD

**Discussion by AngryMooses** [mrlt8/docker-wyze-bridge#762] [Q&A]
I'm planning to feed the video into Scrypted for HKSV, and I was hoping to use in-camera motion detection to trigger HomeKit recordings.  Software motion detection takes a huge CPU hit, is there any way to get the motion passed on from the camera?

**Comment by mrlt8** [comment]
Unfortunately, the cameras don't seem to send motion data back to the app using the TUTK protocol we're using to communicate with the camera. I believe they're using AWS IoT Core to send that data to their servers which then does the push notification to the app.

Older firmware exposed the [SD card over http](https://github.com/mrlt8/docker-wyze-bridge/wiki/Boa-HTTP-Server#motion-alerts) to transfer the timelapse video, so we were able to use that to poll the alarm file that the cameras would send as notification, however, that feature has been removed from all of the firmware (except for the RTSP versions).

I believe some of the other third-party Wyze APIs like [JoshuaMulliken/wyzeapy](https://github.com/JoshuaMulliken/wyzeapy) or [shauntarves/wyze-sdk](https://github.com/shauntarves/wyze-sdk) can poll motion status for each camera from the wyze web API.


Another option would be to do the motion detection on a [sub-stream](https://github.com/mrlt8/docker-wyze-bridge/wiki/Camera-Substreams).

**Comment by mdabbs** [comment]
Do you have access to the uncompressed frames?  I'm assuming you do.  What if you did a simple analysis of the frame looking for a green rectangle somewhere in the image?

**Reply by mrlt8** [reply]
That is one possibility, but the bridge avoids decoding the h264 frames to minimize resources  - you can force it to with FORCE_ENCODE or ROTATE_X.

Another possibility would be try to reverse the smart focus feature as a motion trigger.

**Comment by mdabbs** [comment]
Decoding the substream would be a lot less work than the main, since its just for the purposes of motion detection. Would that be in python? I haven't done much in python, but I could try to contribute.
